engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -296,6 +298,10 @@ private void addReflectionNodes(RenderGraph renderGraph) {
 
         Node worldReflectionNode = new WorldReflectionNode(context);
         renderGraph.addNode(worldReflectionNode, "worldReflectionNode");
+
+        renderGraph.addEdge(renderGraph.findNode(new SimpleUri("engine:shadowMapNode")), reflectedBufferClearingNode);
```

Comment:
Regarding the last comment : Yeah, I just replicated the rendering list, since I didn't want to stare at a black screen and wonder whether the system is flawed or my connections.

Regarding the second last point : The "use the return value of `addNode`" would be best done if we squash all the sub-functions into one large function again, since that way we can communicate the URIs between Nodes.

Which is maybe not the best idea ever, but the best that I can think of right now. Thoughts?

Reply:
I guess you could pass SimpleURIs to the methods called in initRenderGraph(). In the case of the ShadowMapNode you might even place it together with the lighting-related nodes, but it wouldn't help with the more general problem. 

Ultimately it's not easy to get around the fact that by splitting the node instantiations into private methods we created a tree structure  (initRenderGraph() being the root) and a tree cannot fully represent a DAG while the reverse is true. 

I guess if we can get to the point where we can easily produce a diagram of the dependencies/processing order, we might be able to restructure the content of the methods called by initRenderGraph(), perhaps requiring fewer URIs being passed around.